src/host/_stream.cpp

```diff
@@ -142,565 +161,270 @@ using Microsoft::Console::VirtualTerminal::StateMachine;
                                         const til::CoordType sOriginalXPosition,
                                         const DWORD dwFlags,
                                         _Inout_opt_ til::CoordType* const psScrollY)
+try

```

Comment:
It might just be because I'm looking at the diff, but there used to be a lot of comments walking through the code here before. Since this is a pretty high-traffic area, I'd like to see more comments, if possible. At least breaking it down into steps so that it's easier to maintain in the future.

Reply:
I personally didn't find the old comments helpful when making this change, which is why I wrote them from scratch in the new code. But I only documented the parts that I felt like where particularly confusing, which is primarily the backspace handling for tab characters. I avoided commenting something like "Ctrl-X gets written out as ^X in interactive mode" because that's just describing what it does and not why. There's no reason for why it does many of the things the way it does. ☹️ Are there any parts where you felt like a comment would be needed?

Reply:
Sure! Here's a few that would be nice:
- L180: what is "Delayed EOL Wrap" and what we're expected to do when we encounter it (move back to left boundary)
- L193: what is "ENABLE_PROCESSED_OUTPUT" mode? Why do we do `_writeCharsLegacyUnprocessed` instead of the regular one?

That's kinda it, honestly. The switch-case statement is pretty nice for the rest of this function!